Deadmau6 version control

deadmau6 opened the pull request:
Fixes issue #42

deadmau6 committed:
Added maintainers section in README

deadmau6 committed:
Added a basic CHANGELOG template

deadmau6 committed:
versioning

connorjoleary reviewed (commented):
Does this affect the version of the model, lambda, and extension? Or is this a separate version?

deadmau6 commented:
> Does this affect the version of the model, lambda, and extension? Or is this a separate version?

So currently this would affect `VERSIONS_EXTENSION` but I could include the `API`, `MODEL`, and `LAMBDA` to be under a single `VERSION`. Is that something that we want?

connorjoleary commented:
For this project, I am thinking of using a micro-service architecture, with the model, extension, and lambda all being a different service. What I am reading is suggesting that each service should have its own version, so I dont think that we should group them into one.

connorjoleary commented:
But the change log you are suggesting would only apply to the extension right?

deadmau6 commented:
> But the change log you are suggesting would only apply to the extension right?

Correct and if we are doing a `microservices` architecture I agree we should have separate versions and `CHANGELOGS`, so I will get on that. I am gonna be out of town this weekend so I will have to get back to this next week hope that's okay?

connorjoleary commented:
Yes! And thank you so much for putting this together.

deadmau6 committed:
Added DevOps tools for semver

The backend_version.py manages version numbers for backend and the
semver bash script mostly leverages the python script, npm, and git
in order to make things easier.

deadmau6 committed:
Move defaults to a json file

Moving the defaults to a separate json file makes it easier to
manage the version numbers.

deadmau6 committed:
These style changes are from python's json library which formats similar to npm i think

deadmau6 committed:
Updated the semver policy

connorjoleary commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# DeepCite
+# DeepCite ![version](https://img.shields.io/badge/version-2.0.0-blue)
```
Could we hold off on a 2.0.0 version? In my head we are not even at 1.0.0. Also this badge is just for the extension?

deadmau6 replied:
yea it is just for the extension I can specify that, and even add the others

connorjoleary commented on scripts/backend_versions.py:
```diff
@@ -0,0 +1,154 @@
+
+DEFAULT_FPATH = os.path.join(os.getcwd(), '..', 'backend', 'aws', 'lambda', 'defaults.json') 
+PACKAGE_FPATH = os.path.join(os.getcwd(), '..', 'extension','package.json')
+MANIFEST_FPATH = os.path.join(os.getcwd(), '..', 'extension', 'manifest.json') 
```
Thank you for adding these in!

deadmau6 committed:
did a minor update instead

deadmau6 committed:
update badge
